plotly/tools.py

```diff
@@ -2093,14 +2088,717 @@ def get_streamline_arrows(self):
+            kwargs.setdefault('name', 'Decreasing')
+
+            hidden_bar_decr = dict(type='bar',
+                                   x=decrease_x,
+                                   y=decrease_close,
+                                   marker=Marker(color='rgba(0, 0, 0, 0)'),
+                                   legendgroup='Decreasing',
+                                   showlegend=False,
+                                   hoverinfo='none')
+            candle_bar_decr = dict(type='bar',
+                                   x=decrease_x,
+                                   y=decrease_dif,
+                                   legendgroup='Decreasing',
+                                   showlegend=False,
+                                   hoverinfo='none',
+                                   **kwargs)
+            candle_line_decr = dict(type='scatter',
+                                    x=stick_decrease_x,
+                                    y=stick_decrease_y,
+                                    mode='lines',
+                                    legendgroup='Decreasing',
+                                    showlegend=False,
+                                    text=('Low', 'Close', 'Open',
+                                          'High', '') * len(decrease_x),
+                                    **kwargs)
+
+            return hidden_bar_decr, candle_bar_decr, candle_line_decr
+
+        layout = dict(barmode='stack',
+                      yaxis=dict(range=[(min(low) -
+                                        ((max(high) - min(low)) * .1)),
+                                        (max(high) +
+                                        ((max(high) - min(low)) * .1))]))
+
+        if direction is 'increasing':
+            (hidden_bar_incr,
+             candle_bar_incr,
+             candle_line_incr) = make_increasing_candle(increase_x,
+                                                        increase_open,
+                                                        increase_dif,
+                                                        stick_increase_y,
+                                                        stick_increase_x,
+                                                        **kwargs)
+
+            data = [hidden_bar_incr, candle_bar_incr, candle_line_incr]
+
+        elif direction is 'decreasing':
+            (hidden_bar_decr,
+             candle_bar_decr,
+             candle_line_decr) = make_decreasing_candle(decrease_x,
+                                                        decrease_close,
+                                                        decrease_dif,
+                                                        stick_decrease_y,
+                                                        stick_decrease_x,
+                                                        **kwargs)
+            data = [hidden_bar_decr, candle_bar_decr, candle_line_decr]
+
+        elif direction is 'both':
```

Comment:
Similar to https://github.com/plotly/python-api/pull/259/files#r35625061

you can just use an `else` here, then it's clear that `data` will be defined below.